catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/Registry.view.js

```diff
@@ -70,9 +70,42 @@ function (ich,Marionette,_,$,Q,ModalConfig,ModalRegistry,EmptyView,wreqr,Utils,d
             this.model.set('available', this.statusModel.get('value'));
             this.render();
         },
+        //function to called on every RegistryRow when the refresh button is pressed
         updateRemoteStatus: function() {
-            if(this.remoteStatusModel.get('value')){
-                this.model.set('remoteName', this.remoteStatusModel.get('value').remoteName);
+            //collection of configurations which has all models representing every entry in the registryRows.
+            var allConfigurations = this.model.collection.models;
+
+            //ConfigIds has values that are all unique since they cant be duplicated locally (there is a dup check on creation locally)
+            // (only can be dupicated whena remote uses a name that is already used locally
+            var configIds = {};
+            for(var i =0; i<allConfigurations.length; i++){
+                if(!configIds[allConfigurations[i].get('attributes').id]) {
+                    configIds[allConfigurations[i].get('attributes').id] = "";
+                }
+                // else{
+                //     //ERROR! Two configurations found with the same name. Should be impossible. Looking for suggestions on how to display this/what to do on error
+                // }
+            }
+            var currentConfig = this.remoteStatusModel.get('value');
+            //Id's of a config are checked for duplicates before being set and are only set in this method and when first created, after being checked for duplicates.
+            //Therefore we know configIds does not contain duplicates.
+            if(currentConfig && currentConfig.remoteName){
+                //When a config's id !== its remoteName, it's id may need to be updated
+                if(currentConfig.id !== currentConfig.remoteName) {
```

Comment:
Duplicate logic should not be a concern of the ui. It should all be worked out when the configs are created/updated by the back end.